tests/benchmarks/auto/test_auto_benchmark_ppo.py

```diff
@@ -0,0 +1,251 @@
+tasks = benchmarks.get_benchmark('Mujoco1M')['tasks']
+
+
+class TestAutoBenchmarkPPO:
+    """A regression test over PPO Algorithms for automatic benchmarking.
+    (garage-PyTorch-PPO, garage-TensorFlow-PPO, and baselines-PPO2)
+
+    """
+
+    @pytest.mark.benchmark
+    def test_auto_benchmark_ppo_garage_tf(self):
+        # pylint: disable=no-self-use
+        """Create garage TensorFlow PPO model and training.
+
+        Training over different environments and seeds.
+
+        """
+
+        @wrap_experiment
+        def ppo_garage_tf(ctxt, env_id, seed):
+            """Create garage TensorFlow PPO model and training.
+
+            Args:
+                ctxt (garage.experiment.ExperimentContext): The experiment
+                    configuration used by LocalRunner to create the
+                    snapshotter.
+                env_id (str): Environment id of the task.
+                seed (int): Random positive integer for the trial.
+
+            """
+            deterministic.set_seed(seed)
+
+            with LocalTFRunner(ctxt) as runner:
+                env = TfEnv(normalize(gym.make(env_id)))
+
+                policy = TF_GMP(
+                    env_spec=env.spec,
+                    hidden_sizes=(32, 32),
+                    hidden_nonlinearity=tf.nn.tanh,
+                    output_nonlinearity=None,
+                )
+
+                baseline = TF_GMB(
+                    env_spec=env.spec,
+                    regressor_args=dict(
+                        hidden_sizes=(32, 32),
+                        use_trust_region=False,
+                        optimizer=FirstOrderOptimizer,
+                        optimizer_args=dict(
+                            batch_size=32,
+                            max_epochs=10,
+                            tf_optimizer_args=dict(learning_rate=3e-4),
+                        ),
+                    ),
+                )
+
+                algo = TF_PPO(
+                    env_spec=env.spec,
+                    policy=policy,
+                    baseline=baseline,
+                    max_path_length=hyper_parameters['max_path_length'],
+                    discount=0.99,
+                    gae_lambda=0.95,
+                    center_adv=True,
+                    lr_clip_range=0.2,
+                    optimizer_args=dict(
+                        batch_size=32,
+                        max_epochs=10,
+                        tf_optimizer_args=dict(learning_rate=3e-4),
+                        verbose=True))
+
+                runner.setup(algo, env)
+                runner.train(n_epochs=hyper_parameters['n_epochs'],
+                             batch_size=hyper_parameters['batch_size'])
+
+        for env_id, seed, log_dir in benchmark_helper.iterate_experiments(
+                ppo_garage_tf,
+                tasks,
+                seeds,
+                use_tf=True,
+                xcolumn='TotalEnvSteps',
```

Comment:
I think if we call the `csv_to_json` function outside of `iterate_experiments`, we don't have to pass `xcolumn`, `xlabel`, `ycolumn`, and `ylabel` and we can reduce the number of parameters of `iterate_experiments`.

Reply:
The reason I considered to add this to the iterator is because we need to manually call `csv_to_json` for every benchmarking test, which is kinda tedious to me. We need to specify the column and label name anyway, so I'd prefer keeping as is.